### ISSUE TITLE
YouTube embeds not formatting correctly

### ISSUE THREAD

**Issue by mark4409** [friendica/friendica#4957]
When writing a post and clicking the insert video icon, it asks for the URL of the video. If YouTube, then naturally a person would copy and paste the URL into this input box, such as 

> htps://www.youtube.com/watch?v=v8vgNBH3lY8

(note - I've misspelled the url intentionally to prevent it automatically being formatted as a link in here)

This then shows on the message as:

> [video]htps://www.youtube.com/watch?v=v8vgNBH3lY8[/video]

After posting, the video shows up as a small thumbnail with the title floated to the right... which looks a little ugly, and especially on a mobile where that text is squashed into a small area. 

Editing the post shows that the [video] tag has been replaced by the [youtube] tag but still contains the full url, however the youtube BBcode generally only requires the actual code for the video, for example

> [youtube]v8vgNBH3lY8[/youtube]

If I edit the tag so it only contains the video code and repost, then it displays how I would have expected it to.

So... either there needs to be a specific prompt to instruct the user on the correct way of doing this when inserting the link, or there needs to be a backend function that strips the unnecessary part of the URL if its a youtube link, leaving only the youtube video code. 

Tested on 2018-05-dev with Apache 2.4, MariaDB 10.2, and PHP 7.2-FPM

**Comment by MrPetovan** [comment]
The small thumbnail and title is intended to prevent loading the full Youtube iframe with its Javascript and trackers if the user doesn't want to watch the video.

There shouldn't be any difference in display between `[video]https://www.youtube.com/watch?v=v8vgNBH3lY8[/video]`, `[youtube]https://www.youtube.com/watch?v=v8vgNBH3lY8[/youtube]` and `[youtube]v8vgNBH3lY8[/youtube]`

**Comment by mark4409** [comment]
If the small thumbnail is intentional to prevent tracking then fair enough, but the display looks poor on a mobile screen. Perhaps the thumbnail should be larger and the text displayed underneath the image? When floated to the right of the image, a long title wraps every couple of words. Wouldn't it be a better and a more seamless experience to replicate the appearance of the iframe?

**Comment by MrPetovan** [comment]
We probably can improve the display of the thumbnail + title but I don't think we should match the appearance of the Youtube iframe since it requires an extra click to be active and actually be the Youtube player.

**Comment by mark4409** [comment]
Out of interest - does the same behavior apply on a video source that doesn't have the same tracking worries such as a peertube video?

**Comment by MrPetovan** [comment]
Nope, this is specific to YouTube. Video files are shown in the browser-native media player. For PeerTube, it's different since there's no embed code that I know of, so we can only show a link so far.

**Comment by mark4409** [comment]
Just been trying to post another YouTube video this morning. The initial hiding of the iframe only seems to work if the url is specifically in the format `https://www.youtube.com/....`. It doesn't work with the www left out `https://youtube.com/....` or with the mobile version `https://m.youtube.com/....`

**Comment by MrPetovan** [comment]
Thanks for the tests, it will definitely prove useful when I go over the code.

**Comment by miqrogroove** [comment]
Unsure if related, but this type of post never shows up on the Videos tab of the profile.  And there is no upload link on the Videos tab.  So why do I have a Videos tab?

**Comment by MrPetovan** [comment]
You can upload video in the file browser, and thy will appear in this tab, but I agree it’s very counter-intuitive.

**Comment by hoergen** [comment]
Is this still open?

**Comment by MrPetovan** [comment]
That what it says at the top, and no merged PR has been referencing this issue, so the confidence level is pretty high.

**Comment by hoergen** [comment]
Is this still an issue that will be fixed?
